README.md

```diff
@@ -104,6 +104,166 @@ TODO: specify the encoding (byte-array to string) procedure
 
 TODO: specify the decoding (string to byte-array) procedure
 
+### Codecs
+
+Depending on the protocol of a Multiaddr item, different algorithms are used to
+convert their values from/to binary representation. The name of the codec to use
+for each protocol is noted in [protocols.csv](protocols.csv).
+
+In general empty values in the string representation are always disallowed unless
+explicitely noted otherwise. In case of conversion errors implementation must
+refuse to process the given string/binary value and report the error to the caller
+instead.
+
+While all codec types may be encoded using the standard variable length encoding,
+some may also allow being encoded into a specific static-length binary value
+without the extra length information if this is noted in the respective codec's
+description.
+
+All code examples are written in Python-based pseudo code and are optimized for
+legibility rather than speed. In general you should always use existing libraries
+and functions for performing the below conversions rather than rolling your own.
+
+#### `fspath`
+
+Encodes the given Unicode string using the system's local file system encoding.
+On Windows this encoding likely being UTF-16, while being UTF-8 on most other
+systems. It is up to the library to figure out the best encoding value for
+these kinds of strings.
+
+ * String → Binary: `str.encode(SYSTEM_FILESYSTEM_ENCODING)`
+ * Binary → String: `bytes.decode(SYSTEM_FILESYSTEM_ENCODING)`
+
+Protocols using the `fspath` encoding must not be shared between different hosts.
```

Comment:
Go and JS use utf8 and forward slashes. It's up to the application to convert paths into the local operating system's preferred format.